receiver/otlpreceiver/otlphttp.go

```diff
@@ -36,7 +38,20 @@ func handleTraces(resp http.ResponseWriter, req *http.Request, tracesReceiver *t
 
 	otlpResp, err := tracesReceiver.Export(req.Context(), otlpReq)
 	if err != nil {
-		writeError(resp, encoder, err, http.StatusInternalServerError)
+		httpStatus := http.StatusInternalServerError // the default status on errors
+
+		// perhaps we had a gRPC exporter for this data returning an error?
+		if s, ok := status.FromError(err); ok {
+			httpStatus = gRPCToHTTP(s)
+		}
+
+		// perhaps it was an HTTP exporter that failed?
+		reqErr := &colerrs.RequestError{}
+		if errors.As(err, reqErr) {
```

Comment:
I am not sure what is the standard practice now within go, but having something like:
```go
colerrs.IsRequestError(err)
```

Potential saves an allocation?

Reply:
We will need to do an allocation anyway, since we want to create the `reqErr` struct to read the status code from it.